File: leetcode/1653. Minimum Deletions to Make String Balanced/main.go
package main

func minimumDeletions(s string) int {
  n := len(s)
  b := 0

  f := make([]int, n + 1)

  for i := 1; i <= n; i++ {
    if s[i-1] == 'b' {
      f[i] = f[i-1]
      b++
    } else {
      f[i] = min(f[i-1] + 1, b)
    }
  }

  return f[n]
}

func min(a, b int) int {
  if a < b {
    return a
  }

  return b
}

File: leetcode/1653. Minimum Deletions to Make String Balanced/main.rs
impl Solution {
    pub fn minimum_deletions(s: String) -> i32 {
        let n = s.len();

        let chars: Vec<char> = s.chars().collect();
        let mut f = vec![0; n+1];
        let mut b = 0;

        for i in 1..=n {
            if chars[i-1] == 'b' {
                f[i] = f[i-1];
                b += 1;
            } else {
                f[i] = (f[i-1] + 1).min(b);
            }
        }

        f[n]
    }
}

